### PR TITLE
141 resolution tutorial

### LINKED ISSUE #4: Not currently using resolution column
If a resolution column is offered in an ort file, then this should be used in the analysis

### LINKED ISSUE #47: Enable arbitrary functional resolution
Not sure if any packages support this, but it might be cool

### LINKED ISSUE #110: Resolution smearing
Implementation issues:
#127 
#128 
#130 

Currently, we only support a constant Gaussian q-resolution function. However, many reflectometry analysis libraries support q-dependent Gaussian resolution and even arbitrary function resolutions.

https://github.com/easyScience/EasyReflectometryLib/issues/4
https://github.com/easyScience/EasyReflectometryLib/issues/47

The resolution data is a part of the orso format.  The 4th column contains sQ.  When a orso file is read the resolution can be fetched by.  Notice that the sQ is variance and the read value is squared value:
```
from EasyReflectometry.data import load
fpath='.\\tests\\_static\\test_example1.ort'
orso = load(fpath)
np.sqrt(orso['data']['R_spin_up'].variances)
```

### LINKED ISSUE #130: Read resolution smearing from file and use this loaded data
Resolution data should be fetched from (orso) file and passed into the model.

Probably best to make a notbook with an example

### LINKED ISSUE #141: Resolution tutorial

### PR DESCRIPTION
**Pull request by andped10**
Added docs and tutorial to demonstrate how to set a `resolution_function`

Renamed:
- `constant_resolution_function` to `percentage_fhwm_resolution_function`
- `resolution_points` to `fwhm_values`

closes #141 
closes #130 
closes #110 
closes #47
closes #4

### CODE REVIEW

File: docs/experiment/experiment.rst [hunk 1]
```diff
@@ -0,0 +1,81 @@
+Experiment
+==========
+
+The main component of an experiment in :py:mod:`EasyReflectometry` is the `model`. 
+This is a description of the `sample` and the environment in which the experiment is performed. 
+The `Model` is used to calculate the reflectivity of the `sample` at a given set of angles (`q` values).
+The `resolution_functions` are used to account for non-ideal effects in the instrument.
```
Comment (jfkcooper) [inline docs/experiment/experiment.rst:7]: I would reword "The `resolution_functions` are used to account for non-ideal effects in the instrument." to
"The `resolution_functions` are used to quantify the experimental uncertainties in wavelength and angle, allowing the `Model` to accurately describe the data."

File: docs/experiment/experiment.rst [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+:py:func:`percentage_fhwm_resolution_function`
+Often we rely on a resolution function that has a simple functional dependecy of the Q point.
+By this is understood that the applied smearing in an Q point has a FWHM that is given as a percentage of the value of the Q point.
+
+.. code-block:: python 
+
+   from EasyReflectometry.experiment import Model
+   from EasyReflectometry.experiment import percentage_fhwm_resolution_function
+
+   resolution_function = percentage_fhwm_resolution_function(1)
```
Comment (jfkcooper) [inline docs/experiment/experiment.rst:52]: Maybe `1` --> `1.0` to make it more explicit what it is doing

File: docs/tutorials/resolution_functions.ipynb [hunk 3]
```diff
@@ -0,0 +1,380 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "a60117e3-d089-4375-ac7c-12a52ed47271",
+   "metadata": {},
+   "source": [
+    "# Resolution functions\n",
+    "The purpose of the resolution function is to enable the `EasyReflectometry` model to account for various non ideal effects that might be present in the experimental setup.  When determining reflectivity the resolution function defines the smearing to apply.  For a given Q point such smearing is applied by determining an average of the neighboring Q point weigthed by a normal distribution, which has a Q point dependent Full Width at the Half Maximum (FWHM) that again is defined by the resolution function.\n",
```
Comment (jfkcooper) [inline docs/tutorials/resolution_functions.ipynb:9]: "model to account for various non ideal effects that might be present in the experimental setup" --> "model to quantify the experimental uncertainties in wavelength and incident angle."

File: docs/tutorials/resolution_functions.ipynb [hunk 4]
```diff
@@ -0,0 +1,380 @@
...
+    "From the plots it is apparent that an increasing resolution flattens the reflectivity profile."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e3c97d1b",
+   "metadata": {},
+   "source": [
+    "## Afterthoughts\n",
+    "As a last task we will compare the reflectivity determined using a percentage resolution function and a point-wise function.  We should recall that the \"experimental\" data was generated using Refnx.  By comparing the reflectivities determined using a resolution function with a FWHM of 1% and the point-wise FHWN constructed from data in a `.ors` file it is apparent that this reference data also was constructed using a resolution function of 1%."
```
Comment (jfkcooper) [inline docs/tutorials/resolution_functions.ipynb:317]: .ort not .ors

**Review by jfkcooper — CHANGES_REQUESTED** [review]
A few comments on wording to make it slightly more accurately describe the resolution

File: docs/tutorials/resolution_functions.ipynb [hunk 5]
```diff
@@ -0,0 +1,380 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "dad78ccc-1e6f-47cd-8557-c4fa6c736f4b",
+   "metadata": {},
+   "source": [
+    "## Building our model\n",
+    "\n",
+    "The system that was used to produce the data shown above is based on a silicon subphase, with a repeating multilayer of nickel and titanium grown upon it. \n",
```
Comment (jfkcooper) [inline docs/tutorials/resolution_functions.ipynb:125]: The system is not nickel or titanium, it is arbitrary SLD's. We can say it is two metal layers (without being specific), whose SLD's are 4 and 8. It is also not a repeating multilayer, just one layer of each, unless you would describe it being a repeat with zero repeats.

Comment (andped10) [reply docs/tutorials/resolution_functions.ipynb]: Thanks, 
whoops, some copy - paste left behind.
Fixed

File: docs/tutorials/resolution_functions.ipynb (hunk 4 above)
Comment (andped10) [reply docs/tutorials/resolution_functions.ipynb]: done

File: docs/tutorials/resolution_functions.ipynb (hunk 3 above)
Comment (andped10) [reply docs/tutorials/resolution_functions.ipynb]: done

File: docs/experiment/experiment.rst (hunk 2 above)
Comment (andped10) [reply docs/experiment/experiment.rst]: done

File: docs/experiment/experiment.rst (hunk 1 above)
Comment (andped10) [reply docs/experiment/experiment.rst]: done

File: docs/experiment/experiment.rst [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+This will create a :py:class:`Model` instance with the `default_sample` and the environment variables `scale` factor set to 1.0 and a `background` of 1e-6.
+Following the `interface` is set to the default calculator.
+
+
+:py:mod:`resolution_functions`
+------------------------------
+A resolution function enables the `EasyReflectometry` model to account for various non ideal effects that might be present in the experimental setup.
+When determining reflectivity the resolution function defines the smearing to apply.
+For a given Q point such smearing is applied by determining an average of the neighboring Q point weigthed by a normal distribution,
```
[the remark this answers is not in the document]
Comment (andped10) [reply docs/experiment/experiment.rst]: done

File: docs/experiment/experiment.rst [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+   interface = CalculatorFactory()
+   model.interface = interface
+
+This will create a :py:class:`Model` instance with the `default_sample` and the environment variables `scale` factor set to 1.0 and a `background` of 1e-6.
+Following the `interface` is set to the default calculator.
+
+
+:py:mod:`resolution_functions`
+------------------------------
+A resolution function enables the `EasyReflectometry` model to account for various non ideal effects that might be present in the experimental setup.
```
[the remark this answers is not in the document]
Comment (andped10) [reply docs/experiment/experiment.rst]: done

**Review by jfkcooper — APPROVED** [review]
